.github/script-missing-mode-enforcement.py

```diff
@@ -0,0 +1,62 @@
+"""
+Script to scan the OpenTTD's script API for functions that miss checks for the
+function being called from the right mode (deity or company mode).
+
+When a function calls either ScriptObject::Command or ScriptObject::GetCompany
+then the function is considered dangerous. When one of the mode enforcement
+macros from script_error.hpp, i.e. EnforceDeityMode, EnforceCompanyModeValid or
+EnforceDeityOrCompanyModeValid, are called in the function, then we consider
+that the function has mode enforcement.
+
+Any dangerous function for which no enforcement is found are emitted as errors.
+"""
+
+import glob
+import re
+import sys
+
+def checkModeEnforcement(path):
+	errors = []
+	with open(path, 'r') as reader:
+		modeEnforcementFound = False
+		dangerousFunction = False
+		for line in reader:
+			# Line does not start with a tab and have <word>::<word>. That looks like the begin of a function, so reset the state.
+			if re.match("^[^\t].*\w::\w", line):
+				modeEnforcementFound = False
+				dangerousFunction = False
+				currentFunction = line
+			else:
+				if re.match("\t(EnforceDeityMode|EnforceCompanyModeValid|EnforceDeityOrCompanyModeValid|EnforceDeityOrCompanyModeValid_Void)\\(", line):
```

Comment:
It is wise to use `r""` for regular expressions. This is a "raw" string, and avoids weird things like escaping etc. Not required, just so you are aware that exists.